src/qililab/instruments/qdevil/qdevil_qdac2.py

```diff
@@ -65,37 +66,45 @@ def setup(self, parameter: Parameter, value: float | str | bool, channel_id: int
         """
         self._validate_channel(channel_id=channel_id)
 
-        channel = self.device.channel(channel_id)
+        if self.is_device_initialized():
+            channel = self.device.channel(channel_id)
+        else:
+            channel = None
+
```

Comment:
You can write this in one-liner for constistency.


```suggestion
    channel = self.device.channel(channel_id) if self.is_device_initialized() else None

```

Reply:
Sure! Commiting it!